src/autoscaler/api/publicapiserver/public_api_handler.go

```diff
@@ -386,3 +391,66 @@ func (h *PublicApiHandler) GetApiInfo(w http.ResponseWriter, r *http.Request, va
 func (h *PublicApiHandler) GetHealth(w http.ResponseWriter, r *http.Request, vars map[string]string) {
 	w.Write([]byte(`{"alive":"true"}`))
 }
+
+func (h *PublicApiHandler) CreateCustomMetricsCredential(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	appId := vars["appId"]
+	if appId == "" {
+		h.logger.Error("AppId is missing", nil, nil)
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad Request",
+			Message: "AppId is required",
+		})
+		return
+	}
+
+	h.logger.Info("Create custom metric credential", lager.Data{"appId": appId})
+	cred, err := custom_metrics_cred_helper.CreateCustomMetricsCredential(appId, h.policydb, custom_metrics_cred_helper.MaxRetry)
+	if err != nil {
+		h.logger.Error("Failed to create custom metric credential", err, lager.Data{"appId": appId})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Error creating custom metric credential"})
+		return
+	}
+	handlers.WriteJSONResponse(w, http.StatusOK, struct {
+		AppId string `json:"app_id"`
+		*models.CustomMetricCredentials
+		Url string `json:"url"`
+	}{
+		AppId:                   appId,
+		CustomMetricCredentials: cred,
+		Url:                     h.conf.MetricsForwarder.MetricsForwarderUrl,
+	})
+
+}
+
+func (h *PublicApiHandler) DeleteCustomMetricsCredential(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	appId := vars["appId"]
+	if appId == "" {
+		h.logger.Error("AppId is missing", nil, nil)
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad Request",
+			Message: "AppId is required",
+		})
+		return
+	}
+
+	h.logger.Info("Delete custom metric credential", lager.Data{"appId": appId})
+	err := custom_metrics_cred_helper.DeleteCustomMetricsCredential(appId, h.policydb, custom_metrics_cred_helper.MaxRetry)
+	if err != nil {
+		h.logger.Error("Failed to delete custom metric credential", err, lager.Data{"appId": appId})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Error deleting custom metric credential"})
+		return
+	}
+	handlers.WriteJSONResponse(w, http.StatusOK, nil)
+
+}
+
+func (h *PublicApiHandler) Reject(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	handlers.WriteJSONResponse(w, http.StatusForbidden, models.ErrorResponse{
+		Code:    "Forbidden",
+		Message: "Can not take custom metric credential operation in service-offering mode"})
```

Comment:
This command is only valid for build-in auto-scaling capacity.  Please operate service credential with  "cf bind/unbind-service" command.